### ISSUE TITLE
Remove BatchNorm layers once the training is completed.

### ISSUE THREAD

**Issue by zwx8981** [pytorch/pytorch#9678]
In PyTorch, is there any way to remove the batchnorm layers by merging their parameters into their preceding convolutional layers?

ps: If you are familiar with the model deployment function in MatConvNet, you should know what I mean :)

**Comment by zou3519** [comment]
This could be an interesting JIT pass.

But it might be easier to do this by hand right now, @zwx8981

**Comment by Sudhakar17** [comment]
@zou3519 : Can you please explain how to remove BN and it should be mathematically correct? This stack overflow post directly gives b_new and w_new.

https://stackoverflow.com/questions/49536856/tensorflow-how-to-merge-batchnorm-into-convolution-for-faster-inference

**Comment by ahmadr75** [comment]
> In PyTorch, is there any way to remove the batchnorm layers by merging their parameters into their preceding convolutional layers?
> 
> ps: If you are familiar with the model deployment function in MatConvNet, you should know what I mean :)
> 
> cc @albanD @mruberry @jbschlosser

I am currently struggling with the same issue, I actually want to remove BN layers and perform another fine-tuning. Did you reach a better idea than making the network from scratch without BN layers?

**Comment by zou3519** [comment]
cc @jamesr66a could be an interesting FX use case ("remove BatchNorm Layers")

**Comment by jbschlosser** [comment]
See also https://github.com/pytorch/pytorch/issues/58846
